app/Concerns/HasRatings.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+namespace App\Concerns;
+
+use App\Events\User\Updated;
+use App\Libraries\CERT\VATSIMUserDetails;
+use App\Models\Rating;
+use App\Models\RatingPivot;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+trait HasRatings
+{
+    public function ratings(): BelongsToMany
+    {
+        return $this->belongsToMany(
+            Rating::class,
+            "user_ratings",
+            'user_id',
+            'rating_id'
+        )->using(RatingPivot::class)
+            ->wherePivot('deleted_at', '=', null)
+            ->withPivot('deleted_at')
+            ->withTimestamps();
+    }
+
+    public function syncRatings(int $atcRatingCode, int $pilotRatingCode): void
+    {
+        $ratings = collect();
+
+        // Handle ATC rating
+        if ($atcRatingCode === 0) {
+            $this->banNetwork('Network ban discovered via Cert login.');
+            return;
+        }
+
+        // Check if has non-ATC primary rating
+        if ($atcRatingCode >= 8) {
+            $previousRating = VATSIMUserDetails::getPreviousRatingsInfo($this->id);
+            if ($previousRating && isset($previousRating->PreviousRatingInt) && $previousRating->PreviousRatingInt > 0) {
+                // Push pre-special rating
+                $ratings->push(Rating::atcRatingFromID($previousRating->PreviousRatingInt));
+            }
+            // Check if we are going to be adding a different "special" rating than current
+            if($this->specialRating->id != $atcRatingCode){
+                $pivot = $this->specialRating->pivot;
+                $pivot->deleted_at = Carbon::now();
+                $pivot->save();
+            }
+        } else {
+            // Remove any "special" ATC ratings
+            if($this->specialRating){
+                $pivot = $this->specialRating->pivot;
+                $pivot->deleted_at = Carbon::now();
+                $pivot->save();
+            }
+        }
+
+        $this->endNetworkBanIfHas();
+        // Push current rating
+        $ratings->push(Rating::atcRatingFromID($atcRatingCode));
+
+        // Handle pilot ratings
+        for ($i = 1; $i <= 256; $i *= 2) {
+            if ($i & $pilotRatingCode) {
+                $ratings->push(Rating::typePilot()->networkValue($i)->first());
+            }
+        }
+
+        $ratingIds = $ratings->pluck('id')->filter(function ($value) {
+            return $value != null;
+        });
```

Comment:
Bit confused here... you're filtering the ratings based upon their ID but only returning the colelction if they're not null? Surely you'd never get any null ones in the original collection?

Reply:
I believe this may have been left over from previous testing versions when sometimes it wasn't finding a rating (due to seeding issues) - well caught I guess 🖌